backend/controllers/authController.js

```diff
@@ -1,6 +1,23 @@
+/* eslint-disable no-underscore-dangle */
 import jwt from 'jsonwebtoken';
+import bcrypt from 'bcryptjs';
+
 import User from '../models/User.js';
 import sendResetEmail from '../utilities/email.js';
+import google from '../utilities/google.js';
+import fortytwo from '../utilities/42.js';
+
+let userToken = null;
```

Comment:
It could be a problem (in real projects), saving user token on the server. Because at the same time there could be several users trying to login through google or 42. And they could log in as another user.  I actually opened 2 browser windows (one in incognito mode) and tried login in from different accounts at the same time. Several times it came out that I logged in as another user.

I tried to find an approach to how to avoid that, but with no success. We can discuss this on Sunday, maybe for our project this is enough.

Reply:
You are right! I can think of at least some approaches
1) store token in an array or object and send the index/key to front: instead of `auth=token` `auth=<index/key>`
2) send the token to front: `auth=token` `auth=<token>` I don't think this is ok
3) use cookies. This was my original approach, but I think it requires cors to be set `cors({ credentials: true })` which can lead to problems elsewhere.